### PR TITLE
v1.0 Dependency version range updates

### PR DESCRIPTION
**Pull request by prateekdesai04**
*Description of changes:*

Updating caps for numpy, pandas (no change), scikit-learn, CatBoost, XGBoost, LightGBM.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: core/src/autogluon/core/_setup_utils.py [hunk 1]
```diff
@@ -17,9 +17,9 @@
 # Only put packages here that would otherwise appear multiple times across different module's setup.py files.
 DEPENDENT_PACKAGES = {
     "boto3": ">=1.10,<2",  # <2 because unlikely to introduce breaking changes in minor releases. >=1.10 because 1.10 is 3 years old, no need to support older
-    "numpy": ">=1.21,<1.27",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
+    "numpy": ">=1.21,<1.29",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
     "pandas": ">=2.0.0,<2.2.0",  # "<{N+1}" upper cap
-    "scikit-learn": ">=1.3.0,<1.5",  # "<{N+1}" upper cap
+    "scikit-learn": ">=1.3.0,<1.4",  # "<{N+1}" upper cap
```
Comment (Innixma) [inline core/src/autogluon/core/_setup_utils.py:22]: Keep `<1.5` and change comment to `"<{N+2}" upper cap`

scikit-learn follows deprecation strategy where it won't introduce breaking changes between minor versions without warnings, so N+2 is safe.

Comment (prateekdesai04) [reply core/src/autogluon/core/_setup_utils.py:22]: Got it

File: tabular/setup.py [hunk 2]
```diff
@@ -33,7 +33,7 @@
 
 extras_require = {
     "lightgbm": [
-        "lightgbm>=3.3,<3.4",
+        "lightgbm>=3.3,<4.2",  # {N+1} upper cap, where N is the latest released minor version
```
Comment (Innixma) [inline tabular/setup.py:36]: nit comment edit: `"<{N+1}"`

File: tabular/setup.py [hunk 3]
```diff
@@ -46,7 +46,7 @@
     #  It is possibly only present on MacOS, haven't tested linux.
     # XGBoost made API breaking changes in 1.6 with custom metric and callback support, so we don't support older versions.
     "xgboost": [
-        "xgboost>=1.6,<1.8",
+        "xgboost>=1.6,<2.1",  # {N+1} upper cap, where N is the latest released minor version
```
Comment (Innixma) [inline tabular/setup.py:49]: nit comment edit: `"<{N+1}"`

File: core/src/autogluon/core/_setup_utils.py [hunk 4]
```diff
@@ -17,9 +17,9 @@
...
     "scipy": ">=1.5.4,<1.12",  # "<{N+2}" upper cap
```
Comment (Innixma) [inline core/src/autogluon/core/_setup_utils.py:23]: scipy is `N+2`, so upper should be changed to `<1.13` (1.11.3 is latest release)

File: tabular/setup.py [hunk 5]
```diff
@@ -46,7 +46,7 @@
     #  It is possibly only present on MacOS, haven't tested linux.
     # XGBoost made API breaking changes in 1.6 with custom metric and callback support, so we don't support older versions.
     "xgboost": [
-        "xgboost>=1.6,<1.8",
+        "xgboost>=1.6,<2.1",  # <{N+1} upper cap, where N is the latest released minor version
```
Comment (tonyhoo) [inline tabular/setup.py:49]: This might be a major version change, can you verify locally if there is any code change required

**Review by tonyhoo — APPROVED** [review]
LGTM!

**Pull request merged**